src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+        this.validationStringency = validationStringency;
+        this.errorProbabilityAdjustmentFactor = ParamUtils.isPositive(errorProbabilityAdjustmentFactor,
+                "Error adjustment factor must be positive.");
+
+        /* the default Het prior */
+        useBalancedHetPrior();
+    }
+
+    /**
+     * use the BALANCED prior for Het sites.
+     */
+    public void useBalancedHetPrior() {
+        hetPriorType = HetPriorType.BALANCED;
+        logger.info("Using the BALANCED Het prior.");
+    }
+
+    /**
+     * use the HETEROGENEOUS prior for Het sites.
+     * @param minAbnormalFraction estimated minimum fraction of non-germline cells in the sample
+     * @param maxAbnormalFraction estimated maximum fraction of non-germline cells in the sample
+     * @param maxCopyNumber estimated maximum copy number in non-germline events (note: we use a flat probability
+     *                      function for the copy number. using a large value of maxCopyNumber will result in a
+     *                      significant spread of the minor allele fraction prior around 1/2. it is recommended not
+     *                      to use values > 4).
+     * @param quadratureOrder the order of quadrature used in numerical integrations
+     */
+    public void useHeterogeneousHetPrior(final double minAbnormalFraction, final double maxAbnormalFraction,
+                                         final int maxCopyNumber, final int quadratureOrder) {
+
+        hetPriorType = HetPriorType.HETEROGENEOUS;
+
+        /* parameters for building the allele ratio prior at Het sites */
+        this.minAbnormalFraction = ParamUtils.inRange(minAbnormalFraction, 0.0, 1.0, "Minimum fraction of abnormal" +
+                " cells must be between 0 and 1.");
+        this.maxAbnormalFraction = ParamUtils.inRange(maxAbnormalFraction, this.minAbnormalFraction, 1.0, "Maximum fraction of abnormal" +
+                " cells must be greater than the provided minimum and less than 1.");
+        this.maxCopyNumber = ParamUtils.isPositive(maxCopyNumber, "Maximum copy number must be positive");
+
+        /* auxiliary member functions */
+        this.minHetAlleleFraction = (1 - this.maxAbnormalFraction) / (this.maxCopyNumber * this.maxAbnormalFraction +
+                2 * (1 - this.maxAbnormalFraction));
+        this.breakpointHetAlleleFraction = (1 - this.minAbnormalFraction) / (this.maxCopyNumber * this.minAbnormalFraction +
+                2 * (1 - this.minAbnormalFraction));
+
+        /* initialize the integration quadrature and calculate the allele fraction prior on the abscissas */
+        initializeIntegrationQuadrature(ParamUtils.isPositive(quadratureOrder - MIN_QUADRATURE_ORDER,
+                "Quadrature order must be greater than " + MIN_QUADRATURE_ORDER) + MIN_QUADRATURE_ORDER);
+        initializeHetAlleleFractionPrior();
+        logger.info("Using the HETEROGENEOUS Het prior.");
```

Comment:
I think is better to leave the logging for the Tool  that use this helping component.